modules/@apostrophecms/asset/index.js

```diff
@@ -152,29 +72,42 @@ module.exports = {
           await fs.remove(buildDir);
           await fs.mkdirp(buildDir);
 
-          for (const [ name, options ] of Object.entries(self.options.builds)) {
-            let rebuild = argv && !argv['check-ui-build'];
+          for (const [ name, options ] of Object.entries(self.builds)) {
+            // If the option is not present always rebuild everything
+            let rebuild = argv && !argv['check-apos-build'];
 
-            let checkTimestamp = false;
+            if (!rebuild) {
+              let checkTimestamp = false;
 
-            if (options.core) {
-              const bundleExists = await fs.pathExists(bundleDir);
+              // Only builds contributing to the apos admin UI (currently just "apos")
+              // are candidates to skip the build simply because package-lock.json is
+              // older than the bundle. All other builds frequently contain
+              // project level code
+              if (options.apos) {
+                const bundleExists = await fs.pathExists(bundleDir);
 
-              if (!bundleExists) {
-                rebuild = true;
-              }
+                if (!bundleExists) {
+                  rebuild = true;
+                }
 
-              if (!process.env.CORE_DEV) {
-                checkTimestamp = await fs.pathExists(`${bundleDir}/${name}-only-timestamp.txt`);
-              }
+                if (!process.env.APOS_DEV) {
+                  checkTimestamp = await fs.pathExists(`${bundleDir}/${name}-only-timestamp.txt`);
+                }
 
-              if (!rebuild && checkTimestamp) {
-                // If we have a UI build timestamp file compare against the app's
-                // package.json modified time.
-                if (await lockFileIsNewer(name)) {
+                if (!rebuild && checkTimestamp) {
```

Comment:
we don't need to check !rebuild again since it's inside a conditional for that already.